# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/tests/bug503399-ref.html
# Commit: 9b15baab6f77
# Full Hash: 9b15baab6f77d09eca2956276f66fee97555e4a1
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-22 09:56:47
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 3: Make `HTMLEditor::CollapseSelectionToEndOfLastLeafNodeOfDocument` and `HTMLEditor::InitEditorContentAndSelection` do nothing if the document is partially editable r=m_kato
#   
#   They and their callees work with the result of `GetRoot()` which is the document
#   element or the body element.  If the body is not editable, `Selection` should
#   not be updated in non-editable region nor `<br>` elements should not be
# ==============================================================================

diff -r 4d1e93c629da -r 9b15baab6f77 layout/base/tests/bug503399-ref.html
--- a/layout/base/tests/bug503399-ref.html	Thu Sep 22 06:24:38 2022 +0000
+++ b/layout/base/tests/bug503399-ref.html	Thu Sep 22 06:27:37 2022 +0000
@@ -29,9 +29,7 @@
       if (done)
         return;
       try {
-        var r = window.getSelection().getRangeAt(0);
-        r.setStart(p.childNodes[0],14);
-        r.setEnd(p.childNodes[0],14);
+        getSelection().collapse(p.childNodes[0], 14);
       } catch (e) {}
       document.documentElement.removeAttribute('class');
       done = true;